js/kucoin.js

```diff
@@ -3103,8 +3118,14 @@ module.exports = class kucoin extends Exchange {
             // 'sequence': 'RECENTLY_EXPIRE_FIRST',  // Cross: 'RECENTLY_EXPIRE_FIRST' or 'HIGHEST_RATE_FIRST'
             // 'seqStrategy': 'RECENTLY_EXPIRE_FIRST',  // Isolated: 'RECENTLY_EXPIRE_FIRST' or 'HIGHEST_RATE_FIRST'
         };
-        const defaultMarginMode = this.safeString2 (this.options, 'defaultMarginMode', 'marginMode', 'cross');
-        const marginMode = this.safeString (params, 'marginMode', defaultMarginMode); // cross or isolated
+        let marginMode = undefined;
+        [ marginMode, params ] = this.handleMarginModeAndParams ('repayMargin', params);
+        if (marginMode === undefined) {
+            marginMode = 'cross'; // cross as default marginMode
+        }
+        if (symbol !== undefined) {
+            marginMode = 'isolated'; // default to isolated if the symbol argument is defined
```

Comment:
You should default to `cross` that's what every other exchange does

Reply:
It does default to cross, but it defaults to isolated if the symbol is defined, I can remove that though